13.0/s2i/bin/assemble

```diff
@@ -150,8 +150,9 @@ function ismgmtup() {
   return 1
 }
 
-# insert settings for HTTP proxy into settings.xml if supplied
+# insert settings for HTTP proxy into maven settings.xml if supplied
 function configure_proxy() {
+  # if http_proxy_host/port is set, prefer that (oldest mechanism)
```

Comment:
was NO_PROXY non existent when these older mechanisms were en vogue?

Reply:
i'm sure no_proxy existed, i think the history here was:

1) maven wanted separate fields for hostname+port
2) thus it seemed logical to just define env vars for hostname + port
3) since we were defining our own vars for that, we defined our own var for non-proxy too

my goal in introducing support for *proxy variables is to detect your intent (whether you're using our proprietary one, or the standard) and assume you've done that globally (ie if you set http_proxy, you've also set no_proxy).  There's some risk in that assumption but i didn't want to clutter this logic unnecessarily and i think it will probably cover most scenarios.

The obvious exception is the username/password vars which are shared between the two codepaths, because there is no *_proxy variable equivalent for that (unless we're supposed to split it out of the http_proxy url).

Reply:
works for me

Reply:
as it turns out, maven uses a different format ( pipe separated values, not comma) so we can't use no_proxy at all.  reverting this to use HTTP_PROXY_NONPROXY_HOSTS as the variable in all cases.